pages/map/index.tsx

```diff
@@ -1,6 +1,148 @@
+                >
+                    <BiStreetView
+                        //@ts-ignore
+                        lat={center.lat}
+                        lng={center.lng}
+                        text="Marker"
+                        size={26}
+                        style={{position:'absolute', transform: 'translate(-50%, -50%)'}}
+                    />
+
+                    {children}
+                </GoogleMapReact>
+                : 
+                <GoogleMapReact
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                </GoogleMapReact>
+            }
+        </div>
+    )
+}
+
+const MapPageContent = ({children}: MapPageContent) => {
+    const [zip, setZip] = useState('');
+    const [userCoords, setUserCoords] = useState({lat: null, lng: null});
+
+    const changeHandler = (e: any): void => {
+        //Sets zip code state from form value
+        e.preventDefault();
+
+        //@ts-ignore
+        setZip(e.target.value);
+    }
+
+    const getUserCoords = (e: any): void => {
+        //Sets user coordinates as state
+        e.preventDefault();
+
+        if(navigator.geolocation){
+            navigator.geolocation.getCurrentPosition((position) => {
+                let userCoords = {
+                    lat: position.coords.latitude,
+                    lng: position.coords.longitude
+                }
+
+                setUserCoords((prev: any) => {
+                    return {
+                        ...prev,
+                        lat: userCoords.lat,
+                        lng: userCoords.lng
+                    }
+                });
+            });
+        }
+    }
+
+    return(
+        <Container>
+            {children}
+
+            <Row className='mb-2'>
+                <Col>
+                    <Form className='d-flex flex-row justify-content-between' onSubmit={e => e.preventDefault()}>
+                        <Form.Control required type='zip' placeholder='Enter Zip Code...' value={zip} onChange={(e: any) => changeHandler(e)}></Form.Control>
+                        <Button type='submit' className='mx-2'><BiSearchAlt/></Button>
+                        <Button type='submit' onClick={(e: any) => getUserCoords(e)}><BiCurrentLocation/></Button>
+                    </Form>
+                </Col>
+            </Row>
+
+            <Row className='mt-2'>
+                <Col>
+                    <GMap
+                        //@ts-ignore
+                        center={userCoords}
+                    />
+                </Col>
+            </Row>
+        </Container>
+    )
+}
+
 const MapPage = () => {
     return(
-        <h1>Pain</h1>
+        <>
+            <Navigation/>
+            <MapPageContent>
+                <Head>
+                    <title>RecycleIT Map</title>
+                    <link
```

Comment:
this bootstrap css import exists twice. I would create a separate react component that abstracts the head content so this isn't repeated everywhere.